azure/azure_kv.go

```diff
@@ -146,10 +169,10 @@ func (az *azureSecrets) PutSecret(
 		return secrets.NoVersion, err
 	}
 
+	valueStr := string(value)
 	t := func() (interface{}, bool, error) {
-		secretResp, err = az.kv.SetSecret(ctx, az.baseURL, secretName, keyvault.SecretSetParameters{
-			Value: to.StringPtr(string(value)),
-		})
+		params := azsecrets.SetSecretParameters{Value: &valueStr}
+		az.kv.SetSecret(ctx, secretName, params, nil)
```

Comment:
```suggestion
		secretResp, err := az.kv.SetSecret(ctx, secretName, params, nil)
``` 

?

Reply:
Also, we too we can return the version number.
```go
secretResp.ID.Version()
```

Reply:
Good point. But No other KMS in the library is returning the version currently. So I think for now, we can keep it as it is and revisit this later.

Reply:
@sp98, first comment is still not addressed.

Reply:
overlooked that comment. Fixed it now.  Didn't see that since rook was not using the response of the `put` operation.